test/functional/test_framework/test_node.py

```diff
@@ -311,23 +311,21 @@ def wait_for_rpc_connection(self, *, wait_for_import=True):
                 self.rpc_connected = True
                 self.url = self.rpc.rpc_url
                 return
-            except JSONRPCException as e:  # Initialization phase
+            except JSONRPCException as e:
+                # Suppress these as they are expected during initialization.
                 # -28 RPC in warmup
-                # -342 Service unavailable, RPC server started but is shutting down due to error
-                if e.error['code'] != -28 and e.error['code'] != -342:
+                # -342 Service unavailable, could be starting up or shutting down
+                if e.error['code'] not in [-28, -342]:
                     raise  # unknown JSON RPC exception
-            except ConnectionResetError:
-                # This might happen when the RPC server is in warmup, but shut down before the call to getblockcount
-                # succeeds. Try again to properly raise the FailedToStartError
-                pass
             except OSError as e:
-                if e.errno == errno.ETIMEDOUT:
-                    pass  # Treat identical to ConnectionResetError
-                elif e.errno == errno.ECONNREFUSED:
-                    pass  # Port not yet open?
-                else:
+                # Suppress similarly to the above JSONRPCException errors.
+                if e.errno not in [ errno.ECONNRESET,     # This might happen when the RPC server is in warmup,
+                                                          # but shut down before the call to getblockcount succeeds.
+                                    errno.ETIMEDOUT,      # Treat identical to ECONNRESET
+                                    errno.ECONNREFUSED ]: # Port not yet open?
                     raise  # unknown OS error
```

Comment:
Nit: this formatting pushes everything out of my screen:
```suggestion
                # Suppress similarly to the above JSONRPCException errors.
                if e.errno not in [
                    errno.ECONNRESET,  # This might happen when the RPC server is in warmup, but shut down before the call to getblockcount succeeds.
                    errno.ETIMEDOUT,   # Treat identical to ECONNRESET
                    errno.ECONNREFUSED # Port not yet open?
                ]: 
                    raise  # unknown OS error
```

Reply:
Fair. Taken.